common/features/nameService/helpers.ts

```diff
@@ -24,39 +24,12 @@ export function* makeEthCallAndDecode({ to, data, decoder }: Params): SagaIterat
 //#endregion Make & Decode
 
 //#region Mode Map
-function* nameStateOwned({ deedAddress }: IDomainData<NameState.Owned>, nameHash: string) {
-  const ensAddresses = yield select(getENSAddresses);
-
-  // Return the owner's address, and the resolved address if it exists
-  const { ownerAddress }: typeof ENS.deed.owner.outputType = yield call(makeEthCallAndDecode, {
-    to: deedAddress,
-    data: ENS.deed.owner.encodeInput(),
-    decoder: ENS.deed.owner.decodeOutput
-  });
-
-  const { resolverAddress }: typeof ENS.registry.resolver.outputType = yield call(
-    makeEthCallAndDecode,
-    {
-      to: ensAddresses.registry,
-      decoder: ENS.registry.resolver.decodeOutput,
-      data: ENS.registry.resolver.encodeInput({
-        node: nameHash
-      })
-    }
-  );
-
-  let resolvedAddress = '0x0';
-
-  if (resolverAddress !== '0x0') {
-    const result: typeof ENS.resolver.addr.outputType = yield call(makeEthCallAndDecode, {
-      to: resolverAddress,
-      data: ENS.resolver.addr.encodeInput({ node: nameHash }),
-      decoder: ENS.resolver.addr.decodeOutput
-    });
-
-    resolvedAddress = result.ret;
-  }
-
+function* nameStateOwned({}, domain: string) {
+  const namicorn = new Namicorn({ debug: true, disableMatcher: true });
+  namicorn.use(namicorn.middleware.ens({ url: 'https://mainnet.infura.io/mycrypto' }));
```

Comment:
So we generally don't hardcode jsonrpc urls. We use a client-side node-balancer for interacting with the chain. Normally with these types of interactions we would specify it as a new "Namicorn" API [here](https://github.com/MyCryptoHQ/MyCrypto/tree/develop/common/api) and interact with it in that way.

@nickmccurdy @Sonicrida Do you guys have opinions on maintaining existing structure for things like this?

Reply:
I'm not sure, it sounds like a good idea, but does everything need a load balancer? Maybe this could be extracted in such a way that we could add one later.

Reply:
@blurpesec We can sync with the current node url inside the redux state if that's better. Namicorn is just a wrapper around calls to the blockchain. However, Namicorn just calls one method to resolve all names. It seems excessive to put it in the apis.

Reply:
@nickmccurdy Ideally, we wouldn't want all of ENS functionality to drop because infura has a bad day >.>

@perfect-cents That should be fine :-)

Reply:
We are going to make some major updates on Namicorn soon that will enable us to integrate into the MyCrypto name system. After that we will be ready, for another review.